lib/utils/middlewares/context_middleware.ts

```diff
@@ -1,19 +1,32 @@
+import * as express from "express";
+
+import { Option, option } from "ts-option";
 import { right } from "../either";
 
 import { IRequestMiddleware } from "../request_middleware";
 
-import { IContext, IRequestWithContext } from "azure-function-express-cloudify";
+import { IContext } from "azure-function-express";
+
+const CONTEXT_IDENTIFIER = "context";
+
+export function setAppContext(app: express.Express, context: IContext<{}>): void {
+  app.set(CONTEXT_IDENTIFIER, context);
+}
+
+export function getAppContext<T>(request: express.Request): Option<IContext<T>> {
+  return option(request.app.get(CONTEXT_IDENTIFIER));
+}
 
 /**
  * Returns a request middleware that extracts the Azure request context
  * from the request.
  *
  * @param T The type of the bindings found in the context.
- *
- * TODO: validate that the context is indeed defined, respond with ResponseErrorInternal instead
  */
 export function ContextMiddleware<T>(): IRequestMiddleware<never, IContext<T>> {
```

Comment:
for instance, by looking at this function's signature, you cannot know whether this function throws an exception or even what the exception type is, this is also a problem with `Promise<T>` that doesn't have a type for a failed promise, that's the reason I used `Promise<Either<E, T>>`